Exiled.API/Features/Core/Generic/UnmanagedEnumClass.cs

```diff
@@ -0,0 +1,272 @@
+        public override int GetHashCode() => Value.GetHashCode();
+
+        /// <summary>
+        /// Compares the current instance with another object of the same type and returns
+        /// an integer that indicates whether the current instance precedes, follows, or
+        /// occurs in the same position in the sort order as the other object.
+        /// </summary>
+        /// <param name="other">An object to compare with this instance.</param>
+        /// <returns>
+        /// A value that indicates the relative order of the objects being compared.
+        /// The return value has these meanings: Value Meaning Less than zero This instance precedes other in the sort order.
+        /// Zero This instance occurs in the same position in the sort order as other.
+        /// Greater than zero This instance follows other in the sort order.
+        /// </returns>
+        public int CompareTo(TObject other) => Value.CompareTo(other.Value);
+
+        /// <summary>
+        /// Compares the current instance with another object of the same type and returns
+        /// an integer that indicates whether the current instance precedes, follows, or
+        /// occurs in the same position in the sort order as the other object.
+        /// </summary>
+        /// <param name="obj">An object to compare with this instance.</param>
+        /// <returns>
+        /// A value that indicates the relative order of the objects being compared.
+        /// The return value has these meanings: Value Meaning Less than zero This instance precedes other in the sort order.
+        /// Zero This instance occurs in the same position in the sort order as other.
+        /// Greater than zero This instance follows other in the sort order.
+        /// </returns>
+        public int CompareTo(object obj) =>
+            obj == null ? -1 : obj is TSource value ? Value.CompareTo(value) : obj is TObject derived ? Value.CompareTo(derived.Value) : -1;
+
+        /// <summary>
+        /// Compares the specified object instance with another object of the same type and returns
+        /// an integer that indicates whether the current instance precedes, follows, or
+        /// occurs in the same position in the sort order as the other object.
+        /// </summary>
+        /// <param name="x">An object to compare.</param>
+        /// <param name="y">Another object to compare.</param>
+        /// <returns>
+        /// A value that indicates the relative order of the objects being compared.
+        /// The return value has these meanings: Value Meaning Less than zero This instance precedes other in the sort order.
+        /// Zero This instance occurs in the same position in the sort order as other.
+        /// Greater than zero This instance follows other in the sort order.
+        /// </returns>
+        public int Compare(TObject x, TObject y) => x == null ? -1 : y == null ? 1 : x.Value.CompareTo(y.Value);
```

Comment:
I am not sure about this null check actually

Reply:
Ask MSDN and Roslyn